module/behaviour/skills/HeadBehaviourSoccer/src/HeadBehaviourSoccer.cpp

```diff
@@ -62,66 +54,30 @@ namespace module::behaviour::skills {
-        double tanPhi          = std::tan(screen_angular.y());
-        double x               = 0;
-        double y               = 0;
-        double z               = 0;
-        double denominator_sqr = 1 + tanTheta * tanTheta + tanPhi * tanPhi;
-        // Assume facing forward st x>0 (which is fine for screen angular)
-        x = 1 / std::sqrt(denominator_sqr);
-        y = x * tanTheta;
-        z = x * tanPhi;
-
-        return {x, y, z};
-    }
-
     HeadBehaviourSoccer::HeadBehaviourSoccer(std::unique_ptr<NUClear::Environment> environment)
         : Reactor(std::move(environment)) {
 
         on<Configuration>("HeadBehaviourSoccer.yaml")
             .then("Head Behaviour Soccer Config", [this](const Configuration& config) {
-                log_level = config["log_level"].as<NUClear::LogLevel>();
-
-                lastPlanUpdate     = NUClear::clock::now();
-                timeLastObjectSeen = NUClear::clock::now();
-
-                // Config HeadBehaviourSoccer.yaml
-                fractional_view_padding = config["fractional_view_padding"].as<double>();
-
+                log_level         = config["log_level"].as<NUClear::LogLevel>();
                 search_timeout_ms = config["search_timeout_ms"].as<float>();
-
-                fractional_angular_update_threshold = config["fractional_angular_update_threshold"].as<float>();
-
-                headSearcher.setSwitchTime(config["fixation_time_ms"].as<float>());
-
-                oscillate_search = config["oscillate_search"].as<bool>();
-
-                // Note that these are actually modified later and are hence camelcase
-                ballPriority = config["initial"]["priority"]["ball"].as<int>();
-                goalPriority = config["initial"]["priority"]["goal"].as<int>();
-
-                replan_angle_threshold = config["replan_angle_threshold"].as<float>();
-
-                pitch_plan_threshold = config["pitch_plan_threshold"].as<float>() * M_PI / 180.0f;
-                pitch_plan_value     = config["pitch_plan_value"].as<float>() * M_PI / 180.0f;
-
-                // Load searches:
-                for (const auto& search : config["searches"].config) {
-                    SearchType s(search["search_type"].as<std::string>());
-                    searches[s] = std::vector<Eigen::Vector2d>();
-                    for (const auto& p : search["points"]) {
-                        searches[s].push_back(p.as<Expression>());
-                    }
-                }
+                fixation_time_ms  = config["fixation_time_ms"].as<float>();
+                max_yaw           = config["max_yaw"].as<float>();
+                min_yaw           = config["min_yaw"].as<float>();
+                max_pitch         = config["max_pitch"].as<float>();
+                min_pitch         = config["min_pitch"].as<float>();
+
+                // Create vector of search positions
+                search_positions.push_back(Eigen::Vector2d(min_yaw, max_pitch));
```

Comment:
I wonder if these positions would be better off coming from the config file too?

Reply:
I was thinking this, but in this case you'd then have two places to remember to change one value (min/max values and config search positions) so I'm not sure it's worth it. This was is also more readable so you know where it's going to by just reading this.

Reply:
Yeah, we could load the values directly into the search positions from config with the current search path

Reply:
I am just wondering if we'll want a different head behaviour for goalie in which case we could just give it a different config file, but this is probably fine for now

Reply:
I just discovered that yaml does self referencing, how much do you guys hate this or is it worth a shot?
```
min_yaw:  &min_yaw 0.7
max_yaw: &max_yaw 0.7
min_pitch: &min_pitch -0.1
max_pitch: &max_pitch 0.1
search_positions: [[*min_yaw , *max_pitch ], [*min_yaw , *min_pitch], [0, *min_pitch],  ...... ]
```

Reply:
Is it possible to then initialize a std::vector<Eigen::Vector2d(x,y)> directly from the yaml? or would you have to load each position into the vector individually?

Reply:
Not super famiar with the std::vector library sorry, but I think it should be possible? Something like this though I would imagine?
```
search_positions = config["search_positions"].as<std::vector<Eigen::Vector2d>>()
```

Reply:
The yaml library should be smart enough to do that

Reply:
Awesome, will update.